### PR TITLE
Spot wizard

### PR DESCRIPTION
**Pull request by SteffenHeu**
- enable drag & drop for filenamecomponent
- allow non ims files in maldi spot feature detection #1835 
- add wizard setting for maldi spots

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/builders/WizardBatchBuilder.java [hunk 1]
```diff
@@ -123,6 +123,7 @@ yield switch (ionInterface.group()) {
       }
       case MS1_ONLY -> throw new UnsupportedOperationException(
           "Currently not implemented workflow " + workflowPreset);
+      case DRIED_DROPLET -> new WizardBatchBuilderDriedDroplet(steps);
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/builders/WizardBatchBuilder.java:126]: Maybe we can find a name for the workflow that includes different analysis:
Dried droplet, target plate, microarray. Not sure how to best call it. sample matrix?

File: mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/subparameters/WorkflowDriedDropletWizardParameters.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.io.File;
+import java.util.List;
+
+public final class WorkflowDriedDropletWizardParameters extends WorkflowWizardParameters {
+
+  public static OptionalParameter<FileNameParameter> spotNamesFile = new OptionalParameter<>(
+      new FileNameParameter("Spot-to-Sample name file (csv)", """
+          A file containing spot names and the corresponding sample name.
+          The file must contain labelled columns ('spot' and 'name') and be separated by ';'
+          """, List.of(ExtensionFilters.CSV), FileSelectionType.OPEN), false);
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/subparameters/WorkflowDriedDropletWizardParameters.java:42]: Maybe also replace by ExtensionFilters.CSV_TSV_IMPORT and then support tsv

File: mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameComponent.java [hunk 3]
```diff
@@ -168,4 +173,35 @@ public void setToolTipText(String toolTip) {
...
+      e.consume();
+    });
+    txtFilename.setOnDragDropped(e -> {
+      if (e.getDragboard().hasFiles()) {
+        final List<File> files = e.getDragboard().getFiles();
+        final List<String> patterns = new ArrayList<>();
+
+
+        filters.stream().flatMap(f -> f.getExtensions().stream()).forEach(
+            extension -> patterns.add(extension.toLowerCase().replace("*", "").toLowerCase()));
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameComponent.java:191]: Why not just call stream.map.toList at the end? 
like:
var patterns = filters.stream....map(ext -> ext.toLowerCase().replace...).toList

**Review by robinschmid — DISMISSED** [review]
Looks good to me. 
Can you export the wizard graphic smaller? 
the current ones are 701x551 yours is 4k like 4861x3819

File: mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/subparameters/WorkflowDriedDropletWizardParameters.java (hunk 2 above)
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/subparameters/WorkflowDriedDropletWizardParameters.java:42]: not supported for now

File: mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameComponent.java (hunk 3 above)
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameComponent.java:191]: copied from other component. fixed there too

File: mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/builders/WizardBatchBuilder.java (hunk 1 above)
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/builders/WizardBatchBuilder.java:126]: used target plate

File: mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameComponent.java (hunk 3 above)
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameComponent.java]: great :)

File: mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/subparameters/WorkflowDriedDropletWizardParameters.java (hunk 2 above)
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/modules/tools/batchwizard/subparameters/WorkflowDriedDropletWizardParameters.java]: I looked into the code and it seems it requires a csv with semi colon separation. Might be nice in the future to update to csv and tsv support with regular separator. I made an issue for this to keep track.
https://github.com/mzmine/mzmine/issues/1854

**Comment by robinschmid** [comment]
Looks good to me. I will merge it now. 

But I looked into the code and it seems it requires a csv with semi colon separation for the spot names. Might be nice in the future to update to csv and tsv support with regular separator. I made an issue for this to keep track.
https://github.com/mzmine/mzmine/issues/1854

**Review by robinschmid — APPROVED** [review]

**Pull request merged**